src/lib/transition_frontier/persistent_frontier/database.ml

```diff
@@ -355,11 +357,19 @@ let get_transition t hash =
   let%map transition =
     get t.db ~key:(Transition hash) ~error:(`Not_found (`Transition hash))
   in
-  (* this transition was read from the database, so it must have been validated already *)
-  let (`I_swear_this_is_safe_see_my_comment validated_transition) =
-    External_transition.(Validated.create_unsafe @@ decompose transition)
+  let block =
+    let data = External_transition.decompose transition in
+    {With_hash.data; hash={State_hash.State_hashes.state_hash = hash; state_body_hash = None}}
+  in
+  let parent_hash =
+    block
+    |> With_hash.data
+    |> Mina_block.header
+    |> Mina_block.Header.protocol_state
+    |> Mina_state.Protocol_state.previous_state_hash
   in
-  validated_transition
+  (* TODO: the delta transition chain proof is incorrect (same behavior the daemon used to have, but we should probably fix this?) *)
```

Comment:
Let's log an issue to review this. It's fine to merge as is, but we should review this because it could be hiding a subtle bug (if we serve these proofs to bootstrapping nodes or something).